src/bsp/platform/cc3200/xi_bsp_io_fs_cc3200.c

```diff
@@ -0,0 +1,213 @@
+
+#include <flc_api.h>
+typedef struct sBootInfo
+{
+    _u8 ucActiveImg;
+    _u32 ulImgStatus;
+
+} sBootInfo_t;
+
+static sBootInfo_t sBootInfo;
+#define IMG_ACT_USER1 1
+#define IMG_ACT_USER2 2
+#define IMG_BOOT_INFO "/sys/mcubootinfo.bin"
+
+/* copied from TI SDK's flc.c file */
+static _i32 _ReadBootInfo( sBootInfo_t* psBootInfo )
+{
+    _i32 lFileHandle;
+    _u32 ulToken;
+    _i32 status = -1;
+
+    if ( 0 ==
+         sl_FsOpen( ( _u8* )IMG_BOOT_INFO, FS_MODE_OPEN_READ, &ulToken, &lFileHandle ) )
+    {
+        if ( 0 < sl_FsRead( lFileHandle, 0, ( _u8* )psBootInfo, sizeof( sBootInfo_t ) ) )
+        {
+            status = 0;
+        }
+        sl_FsClose( lFileHandle, 0, 0, 0 );
+    }
+
+    return status;
+}
+
+#endif
+
+_u32 xi_bsp_io_fs_open_flags_to_sl_flags( const uint32_t size,
+                                          const xi_fs_open_flags_t open_flags )
+{
+    switch ( open_flags )
+    {
+        case XI_FS_OPEN_WRITE:
+            return FS_MODE_OPEN_CREATE( size, FS_MODE_OPEN_WRITE );
+        case XI_FS_OPEN_APPEND:
+            return FS_MODE_OPEN_WRITE;
+        case XI_FS_OPEN_READ:
+        default:
+            return FS_MODE_OPEN_READ;
+    }
+}
+
+xi_state_t xi_bsp_io_fs_open( const char* const resource_name,
+                              const uint32_t size,
+                              const xi_fs_open_flags_t open_flags,
+                              xi_fs_resource_handle_t* resource_handle_out )
+{
+    ( void )open_flags;
+
+#ifdef XI_DEBUG__FOR_FIRMWARE_UPDATE_TESTING_PURPOSES
+    _ReadBootInfo( &sBootInfo );
+#endif
+
+    if ( 1 == xi_bsp_fwu_is_firmware( resource_name ) )
+    {
+        /* the resource is firmware, handle with FLC (flc_api.h) */
+
+        if ( 0 != sl_extlib_FlcOpenFile( "/sys/mcuimgA.bin", size, NULL,
+                                         &firmware_file_handle_last_opened,
+                                         FS_MODE_OPEN_WRITE ) )
+        {
+            firmware_file_handle_last_opened = 0;
+            *resource_handle_out             = 0;
+            return XI_FS_OPEN_ERROR;
+        }
+
+        *resource_handle_out = firmware_file_handle_last_opened;
+    }
+    else
+    {
+        /* it's an ordinary file, handle with file io (fs.h) */
+
+        _i32 file_handle = 0;
+
+        if ( 0 != sl_FsOpen( ( _u8* )resource_name,
+                             xi_bsp_io_fs_open_flags_to_sl_flags( size, open_flags ),
+                             NULL, &file_handle ) )
+        {
+            *resource_handle_out = 0;
+            return XI_FS_OPEN_ERROR;
+        }
+
+        *resource_handle_out = file_handle;
+    }
+
+    return XI_STATE_OK;
+}
+
+#define XI_BSP_IO_FS_READ_BUFFER_SIZE 1024
+
+xi_state_t xi_bsp_io_fs_read( const xi_fs_resource_handle_t resource_handle,
+                              const size_t offset,
+                              const uint8_t** buffer,
+                              size_t* const buffer_size )
+{
```

Comment:
very good question! It is NOT used yet. It'll be used in dynamic revision when the device will create a checksum of the filecontent. But right, it isn't used yet, and it is not recommended to have this implementation in this PR, but it is here, hope it works, I'll test it in next story (revision handling).